src/EnergyPlus/AirflowNetwork/src/Solver.cpp

```diff
@@ -86,731 +128,12088 @@ namespace AirflowNetwork {
+                DisSysCompELRData(i).ELR = elr;                                         // Value of effective leakage ratio
+                DisSysCompELRData(i).FlowRate = maxflow * m_state.dataEnvrn->StdRhoAir; // Maximum airflow rate
+                DisSysCompELRData(i).RefPres = dp;                                      // Reference pressure difference
+                DisSysCompELRData(i).FlowExpo = expnt;                                  // Air Mass Flow exponent
+
+                // Add the element to the lookup table, check for name overlaps
+                if (elements.find(thisObjectName) == elements.end()) {
+                    elements[thisObjectName] = &DisSysCompELRData(i); // Yet another workaround
+                } else {
+                    ShowSevereError(m_state, format(RoutineName) + "Duplicated airflow element names are found = " + thisObjectName);
+                    // ShowContinueError(state, "A unique component name is required in both objects " + CompName(1) + " and " + CompName(2));
+                    success = false;
+                }
+
+                ++i;
             }
-            if (AFLOW(i) > 0.0) {
-                state.dataAirflowNetwork->AirflowNetworkLinkSimu(i).FLOW = AFLOW(i);
-                state.dataAirflowNetwork->AirflowNetworkLinkSimu(i).FLOW2 = 0.0;
-            } else {
-                state.dataAirflowNetwork->AirflowNetworkLinkSimu(i).FLOW = 0.0;
-                state.dataAirflowNetwork->AirflowNetworkLinkSimu(i).FLOW2 = -AFLOW(i);
+        }
+
+        // Read AirflowNetwork Distribution system component: duct
+        CurrentModuleObject = "AirflowNetwork:Distribution:Component:Duct";
+        DisSysNumOfDucts = m_state.dataInputProcessing->inputProcessor->getNumObjectsFound(m_state, CurrentModuleObject); // Temporary workaround
+        instances = m_state.dataInputProcessing->inputProcessor->epJSON.find(CurrentModuleObject);
+        if (instances != m_state.dataInputProcessing->inputProcessor->epJSON.end()) {
+            int i = 1;                                     // Temporary workaround
+            DisSysCompDuctData.allocate(DisSysNumOfDucts); // Temporary workaround
+            auto &instancesValue = instances.value();
+            for (auto instance = instancesValue.begin(); instance != instancesValue.end(); ++instance) {
+                auto const &fields = instance.value();
+                auto const &thisObjectName = UtilityRoutines::MakeUPPERCase(instance.key());
+                m_state.dataInputProcessing->inputProcessor->markObjectAsUsed(CurrentModuleObject, instance.key()); // Temporary workaround
+
+                Real64 L{fields.at("duct_length")};
+                Real64 D{fields.at("hydraulic_diameter")};
+                Real64 A{fields.at("cross_section_area")};
+                Real64 e{0.0009};
+                if (fields.find("surface_roughness") != fields.end()) {
```

Comment:
Uhhh ... is this the most efficient way to do this? Isn't doing a `find` followed by an `at` doing two lookups for the same object? Wouldn't it be better to do

```cpp
auto &f = fields.find("surface_roughness");
if (f != fields.end())
   e = f.get<Real64>();
```